tests/functional/test_list.py

```diff
@@ -543,3 +544,21 @@ def test_list_path_multiple(tmpdir, script, data):
     json_result = json.loads(result.stdout)
     assert {'name': 'simple', 'version': '2.0'} in json_result
     assert {'name': 'simple2', 'version': '3.0'} in json_result
+
+
+def test_list_skip_work_dir_pkg(script):
```

Comment:
Have you tried to run the test without making changes and confirm if the test fails? Because in my case #7810, the test passes irrespective of the changes. An issue has also been created regarding this https://github.com/pypa/pip/issues/7864 behavior.

Reply:
Hi @gutsytechster , I just did

```
$ tox -e py tests/functional/test_list.py::test_list_skip_work_dir_pkg -s
GLOB sdist-make: /Users/devesh/pip/setup.py
py inst-nodeps: /Users/devesh/pip/.tox/.tmp/package/1/pip-20.1.dev0.zip
py installed: WARNING: Could not generate requirement for distribution -ip 20.1.dev0 (/Users/devesh/pip/.tox/py/lib/python3.8/site-packages): Parse error at "'-ip==20.'": Expected W:(abcd...),apipkg==1.5,atomicwrites==1.3.0,attrs==19.3.0,cffi==1.14.0,coverage==5.0.4,cryptography==2.8,csv23==0.2,execnet==1.7.1,freezegun==0.3.15,mock==4.0.2,more-itertools==8.2.0,pip==20.1.dev0,pluggy==0.13.1,pretend==1.0.9,py==1.8.1,pycparser==2.20,pytest==3.8.2,pytest-cov==2.8.1,pytest-forked==1.1.3,pytest-rerunfailures==6.0,pytest-timeout==1.3.4,pytest-xdist==1.27.0,python-dateutil==2.8.1,PyYAML==5.3.1,scripttest==1.3,setuptools==46.1.3,six==1.14.0,virtualenv==16.7.10,Werkzeug==0.16.0,wheel==0.34.2
py run-test-pre: PYTHONHASHSEED='4055357156'
py run-test-pre: commands[0] | python -c 'import shutil, sys; shutil.rmtree(sys.argv[1], ignore_errors=True)' /Users/devesh/pip/tests/data/common_wheels
py run-test-pre: commands[1] | python /Users/devesh/pip/tools/tox_pip.py wheel -w /Users/devesh/pip/tests/data/common_wheels -r /Users/devesh/pip/tools/requirements/tests-common_wheels.txt
Collecting setuptools>=40.8.0
  Using cached setuptools-46.1.3-py3-none-any.whl (582 kB)
  Saved ./tests/data/common_wheels/setuptools-46.1.3-py3-none-any.whl
Collecting wheel
  Using cached wheel-0.34.2-py2.py3-none-any.whl (26 kB)
  Saved ./tests/data/common_wheels/wheel-0.34.2-py2.py3-none-any.whl
Skipping setuptools, due to already being wheel.
Skipping wheel, due to already being wheel.
py run-test: commands[0] | pytest --timeout 300 tests/functional/test_list.py::test_list_skip_work_dir_pkg
======================================================== test session starts =========================================================
platform darwin -- Python 3.8.2, pytest-3.8.2, py-1.8.1, pluggy-0.13.1
rootdir: /Users/devesh/pip, inifile: setup.cfg
plugins: timeout-1.3.4, rerunfailures-6.0, forked-1.1.3, cov-2.8.1, xdist-1.27.0
timeout: 300.0s
timeout method: signal
timeout func_only: False
collected 1 item                                                                                                                     

tests/functional/test_list.py F                                                                                                [100%]

============================================================== FAILURES ==============================================================
____________________________________________________ test_list_skip_work_dir_pkg _____________________________________________________

script = <tests.lib.PipTestEnvironment object at 0x110b5e8e0>

    def test_list_skip_work_dir_pkg(script):
        """
        Test that list should not include package in working directory
        """
    
        # Create a test package and create .egg-info dir
        pkg_path = create_test_package_with_setup(script,
                                                  name='simple',
                                                  version='1.0')
        script.run('python', 'setup.py', 'egg_info',
                   expect_stderr=True, cwd=pkg_path)
    
        # List should not include package simple when run from package directory
        result = script.pip('list', '--format=json', cwd=pkg_path)
        json_result = json.loads(result.stdout)
>       assert {'name': 'simple', 'version': '1.0'} not in json_result
E       AssertionError: assert {'name': 'simple', 'version': '1.0'} not in [{'name': 'pip', 'version': '20.1.dev0'}, {'name': 'setuptools', 'version': '46.1.3'}, {'name': 'simple', 'version': '1.0'}]
```

Reply:
Then it LGTM :)

Reply:
Thanks @gutsytechster . Please feel free to approve the PR if it looks good :)